File: _arch/algo/dynamic/climbStairs/climb.py
def climb_stairs(n: int) -> int:
    memo = {1: 1, 2: 2}

    def climb(n):
        if n not in memo:
            memo[n] = climb(n - 1) + climb(n - 2)
        return memo.get(n)

    return climb(n)


def climb(n):
    if n<2:
        return n 
    
    return climb(n - 1) + climb(n - 2)

print(climb_stairs(100))

File: _arch/algo/dynamic/climbStairs/climb.js
let climbStairs = function(n) {
    memo = {1:1,2:2}
    
    function climb(n){
        if ( !(n in memo)){
            memo[n] = climb(n-1) + climb(n-2)
        }
            
        return memo[n] 
    }

    return climb(n)
};